patches/api/0440-Allow-proper-checking-of-empty-item-stacks.patch

```diff
@@ -0,0 +1,90 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Aeltumn <daniel@goossens.ch>
+Date: Mon, 28 Aug 2023 13:41:09 +0200
+Subject: [PATCH] Allow proper checking of empty item stacks
+
+This adds a method to check if an item stack is empty or not. This mirrors vanilla's implementation of the same method.
+
+Also fixes incorrect nullability on getCursor().
+
+diff --git a/src/main/java/org/bukkit/event/inventory/InventoryClickEvent.java b/src/main/java/org/bukkit/event/inventory/InventoryClickEvent.java
+index fe58058f9b5d29388d48115cc81dc48ab08c58c1..ac67a4e321f43d1ede09dafe2daa1f07de8a923f 100644
+--- a/src/main/java/org/bukkit/event/inventory/InventoryClickEvent.java
++++ b/src/main/java/org/bukkit/event/inventory/InventoryClickEvent.java
+@@ -85,7 +85,7 @@ public class InventoryClickEvent extends InventoryInteractEvent {
+      *
+      * @return the cursor ItemStack
+      */
+-    @Nullable
++    @NotNull // Paper - fix nullability
+     public ItemStack getCursor() {
+         return getView().getCursor();
+     }
+diff --git a/src/main/java/org/bukkit/inventory/InventoryView.java b/src/main/java/org/bukkit/inventory/InventoryView.java
+index 77a0cd901038405ed0e267f0432f13deccae3ab5..ae2abcb63ac10c20c5bf5c2497806faf380dd2da 100644
+--- a/src/main/java/org/bukkit/inventory/InventoryView.java
++++ b/src/main/java/org/bukkit/inventory/InventoryView.java
+@@ -210,10 +210,10 @@ public abstract class InventoryView {
+     /**
+      * Get the item on the cursor of one of the viewing players.
+      *
+-     * @return The item on the player's cursor, or null if they aren't holding
+-     *     one.
++     * @return The item on the player's cursor, or an empty stack
++     * if they aren't holding one.
+      */
+-    @Nullable
++    @NotNull // Paper - fix nullability
+     public final ItemStack getCursor() {
+         return getPlayer().getItemOnCursor();
+     }
+diff --git a/src/main/java/org/bukkit/inventory/ItemStack.java b/src/main/java/org/bukkit/inventory/ItemStack.java
+index d15a74c38576c49df61cfab02c70fc5d8c0dd5f7..5f35e9c877203c7b65a039b396260e080a34e4f2 100644
+--- a/src/main/java/org/bukkit/inventory/ItemStack.java
++++ b/src/main/java/org/bukkit/inventory/ItemStack.java
+@@ -26,6 +26,17 @@ import org.jetbrains.annotations.Nullable;
+  * returns false.</b>
+  */
+ public class ItemStack implements Cloneable, ConfigurationSerializable, Translatable, net.kyori.adventure.text.event.HoverEventSource<net.kyori.adventure.text.event.HoverEvent.ShowItem>, net.kyori.adventure.translation.Translatable { // Paper
++
++    // Paper start
++    /**
++     * An empty item stack, consists of an air material and a stack size of 0.
++     *
++     * Any item stack with a material of air or a stack size of 0 is seen
++     * as being empty by {@link ItemStack#isEmpty}.
++     */
++    public static final ItemStack EMPTY = new ItemStack();
```

Comment:
Can't do that since it's mutabke

Reply:
Good point, I guess an EMPTY static value is not necessary, it was more for vanilla parity. Now that I think about it vanilla's empty is also mutable? Well weird.